core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -85,28 +96,27 @@
 
     void execute() throws Exception {
         final String writeConsistencyFailure = checkWriteConsistency ? checkWriteConsistency() : null;
-        final ShardId shardId = primary.routingEntry().shardId();
+        ShardRouting primaryRouting = primary.routingEntry();
```

Comment:
If it makes you happy, sure. I don't think it has much, if any effect on the jvm.